src/Mws/FeedClient.php

```diff
@@ -569,7 +569,7 @@ public function updateOrderFulfillment($amazonOrders, $marketPlaceId = 'default'
         $xmlFeed = $this->createXmlFeed($amazonOrders);
 
         /** @var  \MarketplaceWebService_Model_SubmitFeedResponse $response */
-        $response = $this->submitFeed($xmlFeed, $marketPlaceId, $amazonOrder);
+        $response = $this->submitFeed($xmlFeed, $marketPlaceId, array_pop($amazonOrders));
```

Comment:
Ok. El ambito de la variable no depende sel foreach, con lo que como estaba es mas eficiente.
Así queda mas "bonito" pero tienes el pop extra.

Ref: http://stackoverflow.com/questions/13626812/are-php-variables-declared-inside-a-foreach-loop-destroyed-and-re-created-at-eac